src/webpki.rs

```diff
@@ -108,17 +109,27 @@ pub struct EndEntityCert<'a> {
     inner: cert::Cert<'a>,
 }
 
-impl<'a> EndEntityCert<'a> {
+impl<'a> TryFrom<&'a [u8]> for EndEntityCert<'a> {
+    type Error = Error;
+
     /// Parse the ASN.1 DER-encoded X.509 encoding of the certificate
     /// `cert_der`.
-    pub fn from(cert_der: &'a [u8]) -> Result<Self, Error> {
+    fn try_from(cert_der: &'a [u8]) -> Result<Self, Self::Error> {
         Ok(Self {
             inner: cert::parse_cert(
                 untrusted::Input::from(cert_der),
                 cert::EndEntityOrCA::EndEntity,
             )?,
         })
     }
+}
+
+impl<'a> EndEntityCert<'a> {
+    /// Parse the ASN.1 DER-encoded X.509 encoding of the certificate
+    /// `cert_der`.
+    pub fn from(cert_der: &'a [u8]) -> Result<Self, Error> {
+        EndEntityCert::try_from(cert_der)
+    }
```

Comment:
Sorry I wasn't clear in the earlier review. We should also delete this alternative so that `TryFrom` is the only way to construct a `EndEntityCert`.